sql/src/test/java/io/crate/execution/dml/upsert/ShardUpsertRequestTest.java

```diff
@@ -88,4 +94,16 @@ public void testStreaming() throws Exception {
         assertThat(request, equalTo(request2));
     }
 
+    @Test
+    public void testInvalidVersionNumberInItem() {
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("Item with id '1' and version 9223372036854775807 has exceeded " +
+                                        "the maximum version number of 2147483647");
+        new ShardUpsertRequest.Item(
+            "1",
+            null,
+            null,
+            Long.MAX_VALUE);
+    }
```

Comment:
as said, I think testing the validation on `item.version(long)` is (more) important, can you please add it?